packages/noya-state/src/reducers/application.ts

```diff
@@ -1257,6 +1277,231 @@ export function reducer(
         else draft.selectedThemeStyleGroup = '';
       });
     }
+    case 'setTextColor': {
+      const [, color] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded = layer.style?.textStyle?.encodedAttributes;
+          const attributes = layer.attributedString.attributes[0].attributes;
+
+          if (encoded) {
+            if (!encoded.MSAttributedStringColorAttribute) {
+              encoded.MSAttributedStringColorAttribute = color;
+              return;
+            }
+
+            encoded.MSAttributedStringColorAttribute.alpha = color.alpha;
+            encoded.MSAttributedStringColorAttribute.blue = color.blue;
+            encoded.MSAttributedStringColorAttribute.red = color.red;
+            encoded.MSAttributedStringColorAttribute.green = color.green;
+          }
+          if (attributes.MSAttributedStringColorAttribute) {
+            if (!attributes.MSAttributedStringColorAttribute) {
+              attributes.MSAttributedStringColorAttribute = color;
+              return;
+            }
+
+            attributes.MSAttributedStringColorAttribute.alpha = color.alpha;
+            attributes.MSAttributedStringColorAttribute.blue = color.blue;
+            attributes.MSAttributedStringColorAttribute.red = color.red;
+            attributes.MSAttributedStringColorAttribute.green = color.green;
+          }
+        });
+      });
+    }
+    case 'setTextFontName': {
+      const [, value] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+
+      return produce(state, (draft) => {
+        accessPageLayers(draft, pageIndex, layerIndexPaths).forEach((layer) => {
+          if (layer._class !== 'text') return;
+          const encoded =
+            layer.style?.textStyle?.encodedAttributes
+              .MSAttributedStringFontAttribute.attributes;
+          const attributes =
+            layer.attributedString.attributes[0].attributes
+              .MSAttributedStringFontAttribute.attributes;
+
+          if (!encoded) return;
+
+          const face = encoded.name.split('-')[1]
```

Comment:
We can leave this logic, but don't worry about getting it perfect. We'll need more info from the font file before we can do this 100% correctly